internal/auth/oidc/repository_auth_method.go

```diff
@@ -2,9 +2,218 @@ package oidc
 
 import (
 	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	wrapping "github.com/hashicorp/go-kms-wrapping"
+	"google.golang.org/protobuf/proto"
 )
 
-// upsertAccount will create/update account using claims from the user's ID Token.
-func (r *Repository) upsertAccount(ctx context.Context, authMethodId string, IdTokenClaims map[string]interface{}) (*Account, error) {
-	panic("to-do")
+// Account must implement oplog.Replayable for upsertAccount to work
+var _ oplog.ReplayableMessage = (*Account)(nil)
+
+// Account must implement proto.Message for upsertAccount to work
+var _ proto.Message = (*Account)(nil)
+
+// upsertAccount will create/update account using claims from the user's ID and Access Tokens.
+func (r *Repository) upsertAccount(ctx context.Context, am *AuthMethod, IdTokenClaims, AccessTokenClaims map[string]interface{}) (*Account, error) {
+	const op = "oidc.(Repository).upsertAccount"
+	if am == nil || am.AuthMethod == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "missing auth method")
+	}
+	if IdTokenClaims == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "missing ID Token claims")
+	}
+	if AccessTokenClaims == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "missing Access Token claims")
+	}
+	pubId, err := newAccountId()
```

Comment:
Shouldn't this be using the predictable account IDs? Or is that in a different PR?

Reply:
I was planning to add this in a future PR.  ty for the reminder!